src/libraries/Common/src/SourceGenerators/TypeRef.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.DotnetRuntime.Extensions;
+
+namespace DotnetRuntime.SourceGenerators
+{
+    /// <summary>
+    /// An equatable value representing type identity.
+    /// </summary>
+    [DebuggerDisplay("Name={Name}")]
+    public sealed class TypeRef : IEquatable<TypeRef>
+    {
+        private static readonly SymbolDisplayFormat s_minimalDisplayFormat = new SymbolDisplayFormat(
+            globalNamespaceStyle: SymbolDisplayGlobalNamespaceStyle.Omitted,
+            typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypes,
+            genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+            miscellaneousOptions: SymbolDisplayMiscellaneousOptions.UseSpecialTypes);
+
+        public TypeRef(ITypeSymbol type)
+        {
+            (Name, Namespace, GlobalDisplayString, FullyQualifiedDisplayString, MinimalDisplayString) = GetIdentifierInfo(type);
```

Comment:
For the purposes of specifying a minimal type identity token, are all these fields necessary? It seems you'd only need the fully qualified name (either with or without the `global::` prefix)

Reply:
I could remove them. The config binder generator needs them & they felt appropriate for a shared abstraction, just like `IsInterface` and co..

Reply:
The main concern is forcing all those added allocations when not needed, copying the `TypeKind` is cheap.